python/cudf/cudf/core/frame.py

```diff
@@ -3581,64 +3582,122 @@ def _reindex(
         return self._mimic_inplace(result, inplace=inplace)
 
 
-def _get_replacement_values(to_replace, replacement, col_name, column):
+def _get_replacement_values_for_columns(to_replace, value, columns):
```

Comment:
Could we add type annotations here? And possibly a docstring describing what exactly this function is doing - it's not clear from the name.

Reply:
Yep, Done